src/ODataSourceReader.cs

```diff
@@ -653,33 +640,66 @@ private bool CheckIfEndpointIsReadyForUse(string url)
             {
                 checkUrl += "?$top=1";
             }
+            _logger?.Info($"Checking if endpoint: '{_endpoint.Name}' is ready for use on URL: '{checkUrl}'");
             bool result = false;
             Task task;
             var endpointAuthentication = _endpoint.Authentication;
             if (endpointAuthentication.IsTokenBased())
             {
-                string token = OAuthHelper.GetToken(_endpoint, endpointAuthentication, out Exception exception);
-                if (exception != null)
-                {
-                    throw exception;
-                }
-                task = _httpRestClient.GetAsync(checkUrl, HandleResponse, token);
+                string token = GetToken(_endpoint, endpointAuthentication);
+                task = RetryHelper.RetryOnExceptionAsync<Exception>(10, async () => { _httpRestClient.GetAsync(checkUrl, HandleResponse, token, (Dictionary<string, string>)GetAllHeaders()).Wait(new CancellationTokenSource(_timeoutInMilliseconds).Token); }, _logger);
             }
             else
             {
-                task = _httpRestClient.GetAsync(checkUrl, HandleResponse, endpointAuthentication);
+                task = RetryHelper.RetryOnExceptionAsync<Exception>(10, async () => { _httpRestClient.GetAsync(checkUrl, HandleResponse, endpointAuthentication, (Dictionary<string, string>)GetAllHeaders()).Wait(new CancellationTokenSource(_timeoutInMilliseconds).Token); }, _logger);
+            }
+            if (task.IsCanceled)
+            {
+                string aditionalErrorMSG = _requestTimedOutFromGlobalSettings ? "(To change go to global settings and look for TimeoutInMilliseconds)" : "";
```

Comment:
Add a better explanation of the need for the user to change the timeout in this GS path:
/Globalsettings/Modules/DataIntegration/Job/TimeoutInMilliseconds
Since by default there will be no such "TimeoutInMilliseconds" key in the Swift Global settings file so no-one would know what to update or add :)